src/System Application/App/Extension Management/src/ExtensionManagement.Page.al

```diff
@@ -208,37 +215,33 @@ page 2500 "Extension Management"
                         CurrPage.Update(false);
                     end;
                 }
+#if not CLEAN25
                 action("Extension Marketplace")
                 {
                     Caption = 'Extension Marketplace';
                     Enabled = IsSaaS;
                     Image = NewItem;
                     ToolTip = 'Browse the extension marketplace for new extensions to install.';
-                    Visible = not IsOnPremDisplay;
-#if not CLEAN24                    
-#pragma warning disable AL0432
-                    RunObject = page "Extension Marketplace";
-#pragma warning restore AL0432
-#else                    
+                    Visible = false;
+                    ObsoleteState = Pending;
+                    ObsoleteReason = 'This action will be obsoleted. Microsoft AppSource apps feature will replace the Extension Marketplace.';
+                    ObsoleteTag = '25.0';
+
                     trigger OnAction()
                     begin
                         Hyperlink('https://aka.ms/bcappsource');
                     end;
-#endif
                 }
+#endif
                 action("Microsoft AppSource Gallery")
                 {
                     Caption = 'AppSource Gallery';
                     Enabled = IsSaaS;
                     Image = NewItem;
                     ToolTip = 'Browse the Microsoft AppSource Gallery for new extensions to install.';
                     Visible = not IsOnPremDisplay;
+                    RunObject = Page 2515; // Using ID to avoid circular dependency.
```

Comment:
this creates a logical dependency issue. The way to fix it is to have the AppSource Gallery module add this action using a page extension.